### ISSUE TITLE
Test suite fails to build

### ISSUE THREAD

**Issue by mdorman** [DanielG/cabal-helper#10]
http://hydra.cryp.to/build/1099729/nixlog/2/raw has a full build log, but the pertinent bit is:

```
CabalHelper/Main.hs:206:43:
    Couldn't match expected type ‘NubListR
                                    (InstalledPackageId,
                                     PackageId,
                                     Distribution.PackageDescription.ModuleRenaming)’
                with actual type ‘[a1]’
    In the ‘ghcOptPackages’ field of a record
    In the expression:
      res
        {ghcOptPackageDBs = withPackageDB lbi,
         ghcOptHideAllPackages = Flag True,
         ghcOptPackages = nub $ ghcOptPackages res}
    In an equation for ‘res'’:
        res'
          = res
              {ghcOptPackageDBs = withPackageDB lbi,
               ghcOptHideAllPackages = Flag True,
               ghcOptPackages = nub $ ghcOptPackages res}

CabalHelper/Main.hs:206:49:
    Couldn't match expected type ‘[a1]’
                with actual type ‘NubListR
                                    (InstalledPackageId,
                                     PackageId,
                                     Distribution.PackageDescription.ModuleRenaming)’
    In the second argument of ‘($)’, namely ‘ghcOptPackages res’
    In the ‘ghcOptPackages’ field of a record

CabalHelper/Main.hs:378:48:
    Couldn't match expected type ‘NubListR
                                    (InstalledPackageId,
                                     PackageId,
                                     Distribution.PackageDescription.ModuleRenaming)’
                with actual type ‘[a0]’
    In the ‘ghcOptPackages’ field of a record
    In the expression:
      opts {ghcOptPackages = nub $ ghcOptPackages opts}
    In an equation for ‘nubPackageFlags’:
        nubPackageFlags opts
          = opts {ghcOptPackages = nub $ ghcOptPackages opts}

CabalHelper/Main.hs:378:54:
    Couldn't match expected type ‘[a0]’
                with actual type ‘NubListR
                                    (InstalledPackageId,
                                     PackageId,
                                     Distribution.PackageDescription.ModuleRenaming)’
    In the second argument of ‘($)’, namely ‘ghcOptPackages opts’
    In the ‘ghcOptPackages’ field of a record
```

**Comment by peti** [comment]
I'm getting these errors also during development:

```
$ ghc-mod debug -v9
[1 of 4] Compiling CabalHelper.Types ( CabalHelper/Types.hs, /home/simons/.ghc-mod/cabal-helper/CabalHelper/Types.o )
[2 of 4] Compiling CabalHelper.Common ( CabalHelper/Common.hs, /home/simons/.ghc-mod/cabal-helper/CabalHelper/Common.o )
[3 of 4] Compiling CabalHelper.Sandbox ( CabalHelper/Sandbox.hs, /home/simons/.ghc-mod/cabal-helper/CabalHelper/Sandbox.o )
[4 of 4] Compiling Main             ( CabalHelper/Main.hs, /home/simons/.ghc-mod/cabal-helper/Main.o )

CabalHelper/Main.hs:206:43:
    Couldn't match expected type ‘NubListR
                                    (InstalledPackageId,
                                     PackageId,
                                     Distribution.PackageDescription.ModuleRenaming)’
                with actual type ‘[a1]’
    In the ‘ghcOptPackages’ field of a record
    In the expression:
      res
        {ghcOptPackageDBs = withPackageDB lbi,
         ghcOptHideAllPackages = Flag True,
         ghcOptPackages = nub $ ghcOptPackages res}
    In an equation for ‘res'’:
        res'
          = res
              {ghcOptPackageDBs = withPackageDB lbi,
               ghcOptHideAllPackages = Flag True,
               ghcOptPackages = nub $ ghcOptPackages res}

CabalHelper/Main.hs:206:49:
    Couldn't match expected type ‘[a1]’
                with actual type ‘NubListR
                                    (InstalledPackageId,
                                     PackageId,
                                     Distribution.PackageDescription.ModuleRenaming)’
    In the second argument of ‘($)’, namely ‘ghcOptPackages res’
    In the ‘ghcOptPackages’ field of a record

CabalHelper/Main.hs:378:48:
    Couldn't match expected type ‘NubListR
                                    (InstalledPackageId,
                                     PackageId,
                                     Distribution.PackageDescription.ModuleRenaming)’
                with actual type ‘[a0]’
    In the ‘ghcOptPackages’ field of a record
    In the expression:
      opts {ghcOptPackages = nub $ ghcOptPackages opts}
    In an equation for ‘nubPackageFlags’:
        nubPackageFlags opts
          = opts {ghcOptPackages = nub $ ghcOptPackages opts}

CabalHelper/Main.hs:378:54:
    Couldn't match expected type ‘[a0]’
                with actual type ‘NubListR
                                    (InstalledPackageId,
                                     PackageId,
                                     Distribution.PackageDescription.ModuleRenaming)’
    In the second argument of ‘($)’, namely ‘ghcOptPackages opts’
    In the ‘ghcOptPackages’ field of a record
ghc-mod: readCreateProcess: /nix/store/hfxdl8rp576nb9lh36cp480pjqg4sb76-haskell-cabal-helper-0.5.2.0/libexec/cabal-helper-wrapper "/home/simons/src/cabal2nix" "/home/simons/src/cabal2nix/dist" "package-db-stack" "entrypoints" "source-dirs" "ghc-options" "ghc-src-options" "ghc-pkg-options" "ghc-merged-pkg-options" "ghc-lang-options" "--with-ghc=ghc" "--with-ghc-pkg=ghc-pkg" "--with-cabal=cabal" (exit 1): failed
```

**Comment by alanz** [comment]
According to hackage, cabal-helper-5.2.0 has been deprecated

https://hackage.haskell.org/package/cabal-helper/preferred

"Explicitly deprecated versions for cabal-helper include: 0.1.0.0, 0.5.2.0"

**Comment by mdorman** [comment]
@peti Doh, you're right, it's not just the test suite that was broken---I hadn't noticed since I've been using a version including my patch in #11.  It seems a shame it's been ignored.

**Comment by peti** [comment]
@alanz, how long ago (approximately) was that version marked deprecated? https://github.com/commercialhaskell/all-cabal-files/blob/hackage/preferred-versions doesn't contain that change yet.

**Comment by alanz** [comment]
I'm not sure, probably in the last 10 days some time

**Comment by DanielG** [comment]
I think a day after the release or so, I implemented running the tests against all versions of Cabal that work with a given version of ghc so this shouldn't happen again. I'll release a new version by the end of the week or so, sorry about that.

**Comment by DanielG** [comment]
Should be fixed in 0.5.3.0.

**Comment by peti** [comment]
Thank you very much for the update.
